### PR TITLE
KAFKA-7937: Fix Flaky Test ResetConsumerGroupOffsetTest.testResetOffs…

### PR DESCRIPTION
**Pull request by gwenshap**
Address the comments on PR-6307. Sorry for new PR, but one of the comments was to move the PR to another branch.

***
Since the test fails sometimes on lack of coordinator, I'm giving it a bit more attempts to find it.

I admit that I haven't been able to actually reproduce this failure, so I'm only hoping this fixes it. But it doesn't fail more often than it used to (on my machine)

Fixing on 2.2 because the intent is to fix enough flakes to allow for a clean release.

### CODE REVIEW

**Comment by gwenshap** [comment]
CC @vahidhashemian @mjsax

**Review by vahidhashemian — APPROVED** [review]
LGTM. I'll wait for the build to pass before merging. Thanks!

**Comment by gwenshap** [comment]
Thanks! Don't forget to cherrypick to 2.2 while you are at it :)

**Comment by vahidhashemian** [comment]
Sure. I haven't yet done cherry-picking when merging. That would be my learning exercise :)

**Comment by vahidhashemian** [comment]
The failed unit test seems to be unrelated.

**Comment by vahidhashemian** [comment]
Cherry-picked to 2.2 branch.

**Comment by ijuma** [comment]
Isn't this a bug in the command that it doesn't retry? Seems like this change is hiding a real bug. Also, we should aim to include information in the commit messages themselves, seems like they're empty aside from the title.

**Comment by vahidhashemian** [comment]
Agreed. The retry logic existed in the old `AdminClient`, but it's not there yet in the new one. There is [a JIRA](https://issues.apache.org/jira/browse/KAFKA-6789) open for it though.

Also good point on commit title and message. If they should be there for each commit (which makes sense), I am wondering if there is a good way to enforce them as part of the test or when merging?

**Comment by ijuma** [comment]
Can we add a comment to that open PR stating that we should revert this change there?

For the commit message, it's the responsibility of the committer to ensure the commit message is good. The same way you have to review the code. Personally, I often edit it myself before merging instead of going back to the PR author if the change is simple. For complex changes, I ask the author to update the PR description and I then copy and paste.

**Comment by vahidhashemian** [comment]
Created [KAFKA-7993](https://issues.apache.org/jira/browse/KAFKA-7993) to better track it.
Understood your point about the commit message. Will try to be more diligent about their quality.

**Pull request merged**
